applications/FluidDynamicsApplication/custom_processes/apply_mach_dependent_boundary_conditions.cpp

```diff
@@ -0,0 +1,296 @@
+#include "apply_mach_dependent_boundary_conditions.h"
+#include "fluid_dynamics_application_variables.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/normal_calculation_utils.h"
+namespace Kratos
+{
+
+
+ApplyMachDependentBoundaryConditions::ApplyMachDependentBoundaryConditions(
+    Model& rModel,
+    Parameters Parameters)
+{
+    KRATOS_TRY
+
+    Parameters.ValidateAndAssignDefaults(GetDefaultParameters());
+    mpModelPart = &rModel.GetModelPart(Parameters["model_part_name"].GetString());
+    
+    // Reading flow direction
+    const auto flow_direction_name = Parameters["flow_direction_variable"].GetString();
+
+    KRATOS_ERROR_IF_NOT(KratosComponents<VectorVariable>::Has(flow_direction_name))
+        << "specified flow direction variable is not a known vector variable." << std::endl;
+    
+    mpFlowDirectionVariable = & KratosComponents<VectorVariable>::Get(flow_direction_name);
+
+    mRefreshNormalsEveryTimeStep = Parameters["refresh_normals_every_time_step"].GetBool();
+
+    // Reading boundary conditions to enforce
+    mSubsonicBCs.reserve(Parameters["subsonic_boundary_conditions"].size());
+    mSupersonicBCs.reserve(Parameters["supersonic_boundary_conditions"].size());
+
+    for(auto & var_settings : Parameters["subsonic_boundary_conditions"])
+    {
+        ReadBoundaryCondition(mSubsonicBCs, var_settings);
+    }
+
+    for(auto & var_settings : Parameters["supersonic_boundary_conditions"])
+    {
+        ReadBoundaryCondition(mSupersonicBCs, var_settings);
+    }
+
+    KRATOS_CATCH("");
+}
+
+void ApplyMachDependentBoundaryConditions::ExecuteInitialize()
+{
+    NormalCalculationUtils().CalculateUnitNormals<Condition>(*mpModelPart);
+}
+
+
+void ApplyMachDependentBoundaryConditions::ExecuteInitializeSolutionStep()
+{
+    if(mRefreshNormalsEveryTimeStep)
+    {
+        NormalCalculationUtils().CalculateUnitNormals<Condition>(*mpModelPart);
+    }
+
+    // Enabling and disabling BC according to the time and active interval
+    const double time = mpModelPart->GetProcessInfo().GetValue(TIME);
+
+    for(auto & boundary_condition: mSupersonicBCs)
+    {
+        boundary_condition.ActivateIfInsideTimeInterval(time);
+    }
+
+    for(auto & boundary_condition: mSubsonicBCs)
+    {
+        boundary_condition.ActivateIfInsideTimeInterval(time);
+    }
+
+    // Enforcing boundary conditions
+    block_for_each(mpModelPart->Nodes(), [&](NodeType & rNode)
+    {
+        // Computing mach projected onto normal
+        const auto & normal = rNode.FastGetSolutionStepValue(NORMAL);
+        const auto & flow_direction = rNode.FastGetSolutionStepValue(*mpFlowDirectionVariable);
+
+        const double & mach = rNode.GetValue(MACH);
```

Comment:
```suggestion
        const double mach = rNode.GetValue(MACH);
```
Memory is the same so if it is not strictly required better use a copy.